presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -652,4 +636,15 @@ public static int getFilterAndProjectMinOutputPageRowCount(Session session)
     {
         return session.getSystemProperty(FILTER_AND_PROJECT_MIN_OUTPUT_PAGE_ROW_COUNT, Integer.class);
     }
+
+    private static Integer validateValueIsPowerOfTwo(Object value, String property)
+    {
+        int intValue = ((Number) value).intValue();
```

Comment:
1. `requireNonNull(value, "value is null")`
2. can the value be "more than `int`"? (i.e. can this line do a silent truncation of e.g. a `long`?)

Reply:
It shouldn't be anything other than `Integer`. This is because you specify type of value in `PropertyMetadata`